File: java-service/src/main/java/com/poc/tutorials/repository/TutorialRepository.java
package com.poc.tutorials.repository;

import com.poc.tutorials.model.TutorialEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface TutorialRepository
        extends JpaRepository<TutorialEntity, UUID>, JpaSpecificationExecutor<TutorialEntity> {
    List<TutorialEntity> findByPublished(boolean published);

    List<TutorialEntity> findByTitleContaining(String title);
}

File: kotlin-service/src/main/kotlin/com/example/poc/repository/TutorialRepository.kt
package com.example.poc.repository

import com.example.poc.model.TutorialEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TutorialRepository : JpaRepository<TutorialEntity, UUID>, JpaSpecificationExecutor<TutorialEntity> {
    fun findByPublished(published: Boolean) : List<TutorialEntity>
    fun findByTitleContaining(title: String) : List<TutorialEntity>
}